### ISSUE TITLE
bug(autocomplete): matAutocomplete attribute on input tags should not add role="combobox"

### ISSUE THREAD

**Issue by csirota97-a11y** [angular/components#33401]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Marking an input field as a autocomplete adds `role="combobox"` attribute which violates WCAG ITEM: 4.1.2 Name, Role, Value.

### Reproduction

StackBlitz link: 
Steps to reproduce:
1. https://stackblitz.com/edit/components-issue-starter-xjowe1gm?file=src%2Fmain.ts
2. Use browser console to see that input tag has `role="combobox"`

### Expected Behavior

input tag role should accurately reflect input tag

### Actual Behavior

input tag has `role="combobox"` when autocomplete is used

### Environment

- Angular: 18-22* not tested on earlier versions
- CDK/Material: 18-22
- Browser(s): Edge, Chrome, Safari
- Operating System (e.g. Windows, macOS, Ubuntu): Mac and Windows

**Comment by crisbeto** [comment]
This is the expected behavior. When the autocomplete is open, the trigger gets connected to a `listbox`.
